### PR TITLE
fix: report the real error when newSocketMount() fails

### PR DESCRIPTION
**Pull request by hessjcg**
Formatting the URI into an instance name when reporting an error should not hide the real error. Also, 
avoid port conflict in root_test.go.

### CODE REVIEW

File: internal/proxy/proxy.go [hunk 1]
```diff
@@ -406,8 +406,8 @@ func NewClient(ctx context.Context, d alloydb.Dialer, l alloydb.Logger, conf *Co
 					l.Errorf("failed to close mount: %v", mErr)
 				}
 			}
-			i, err := ShortInstURI(inst.Name)
-			if err != nil {
+			i, instUriErr := ShortInstURI(inst.Name)
```
Comment (enocom) [inline internal/proxy/proxy.go:409]: Why this change? Isn't `err` enough here?

**Review by enocom — COMMENTED** [review]
Thanks for fixing this! We probably needed a smarter approach to port selection as is.

File: internal/proxy/proxy.go (hunk 1 above)
Comment (hessjcg) [reply internal/proxy/proxy.go:409]: The variable `err` is initially declared on line 401: `m, err := newSocketMount(ctx, conf, pc, inst)`. Then if that err is not nil, we need to report why not. 

On line 409, we are attempting to format the name of the instance that caused `newSocketMount()` to return an error. Redeclaring the variable `err` on line 409  hides `err` from line 401. 

Then, on line 414, we are attempting to return an error from this function, including `err` in the message. 

If we redeclare `err` on 409, then the statement on line 414 uses the `err` from line 409 instead of what we wanted:  the error from line 401.

**Review by enocom — APPROVED** [review]
